tic-tac-toe/src/Board.jsx

```diff
@@ -0,0 +1,70 @@
+import React, {useContext} from 'react';
+import Square from './Square';
+import BoardState from './contexts/BoardContext';
+import declareWinner from './contexts/declareWinner';
+
+const Board = (props) => {
+  const { turn, setTurn, boardState, setBoardState, history, setHistory} = useContext(BoardState);
+
+  const winner = declareWinner(boardState);
+  let status;
+  if (winner) {
+    status = `The winner is ${winner}!`;
+  }
+
+  function handleClick(num) {
+    if (!declareWinner(boardState) && !boardState[num]) {
+      const updatedHistory = [...history];
+      const updatedBoard = [...boardState];
+      updatedBoard[num] = turn;
+      updatedHistory.push(updatedBoard);
+      setBoardState(updatedBoard);
+      setHistory(updatedHistory);
+      setTurn((turn === 'O') ? 'X' : 'O');
+    }
+  }
+
+  function stepTo(step) {
+    const boardInHistory = [...history];
+    setBoardState(boardInHistory[step - 1] || []);
+    setHistory(boardInHistory[step - 1] || []);
+    setTurn((step % 2 === 0) ? 'X' : 'O');
+  }
+
+  const playerMoves = history.map((step, move) => {
+    const desc = move ?
+      'Go to move #' + move :
+      'Go to game start';
+    return (
+      <li key={move}>
+        <button onClick={() => stepTo(move)}>{desc}</button>
+      </li>
+    );
+  });
+
+  return (
+    <div className='board'>
+      <h3>{status}</h3>
+      <div className="game-info">
+        <ol>{playerMoves}</ol>
+      </div>
+      <div className='row'>
+        <Square clickEvent={() => handleClick(0)} value={boardState[0]}/>
```

Comment:
I'd probably use `board.slice` to take the first three items, then map over them for this row. You could then do the same thing for the row below and the second three items. Another option would be to use something like the [chunk](https://lodash.com/docs/4.17.15) method from lodash that you could iterate over, render a row, and then render a square for each sub element.